src/main/java/com/powsybl/openloadflow/network/LfNetwork.java

```diff
@@ -621,13 +621,11 @@ public void fixTransformerVoltageControls() {
             } else {
                 // I don't know.
             }
-            if (componentOnNotControlledSide != null) {
-                if (componentOnNotControlledSide.stream().noneMatch(LfBus::isVoltageControlled)) {
-                    branch.setVoltageControlEnabled(false);
-                    LOGGER.trace("Transformer {} voltage control has been disabled because no PV buses on not controlled side connected component",
-                            branch.getId());
-                    disabledTransformerCount++;
-                }
+            if (componentOnNotControlledSide != null && componentOnNotControlledSide.stream().noneMatch(LfBus::isVoltageControlled)) {
```

Comment:
Is there any chance we're looping many times over a big set of LfBus? Then we should save the results for each component.
But it would mean that there is a big (main) connected component with many controller branches with the corresponding controlled buses outside that component. And no other controlled bus. Not very likely I suppose.

Reply:
Yes, it's likely:
typically you will have a very large component for EHV (400-225 kV), on which will be connected most transformers that regulate on the lower level (90-63 kV in France).

Reply:
I added a commit to save the result for each component, it's supposed to be a bit faster... or at least more robust. It seems that in practice it's not faster as in big components, the `noneMatch` encounters always quickly a controlled bus.